persistence/sql/persister_identity.go

```diff
@@ -472,3 +510,65 @@ func (p *Persister) injectTraitsSchemaURL(ctx context.Context, i *identity.Ident
 	i.SchemaURL = s.SchemaURL(p.r.Config(ctx).SelfPublicURL(nil)).String()
 	return nil
 }
+
+func (p *Persister) getJsonSearchQuery(field string, values []string) pop.ScopeFunc {
+	return func(q *pop.Query) *pop.Query {
+		dsn := p.r.Config(context.Background()).DSN()
+		switch {
+		case strings.HasPrefix(dsn, "sqlite"), strings.HasPrefix(dsn, "mysql"):
+			field, innerField := extractFieldAndInnerFields(field)
+			for _, value := range values {
+				if innerField == "" {
+					q = q.Where(fmt.Sprintf(`json_extract(%s, '$') = "%v"`, field, value))
+				} else {
+					q = q.Where(fmt.Sprintf(`json_extract(%s, '$.%v') = "%v"`, field, innerField, value))
+				}
+			}
+			return q
+		case strings.HasPrefix(dsn, "postgres"), strings.HasPrefix(dsn, "crdb"), strings.HasPrefix(dsn, "cockroach"):
+			field, innerField := extractFieldAndInnerFields(field)
+			for _, value := range values {
+				if innerField == "" {
+					q = q.Where(fmt.Sprintf(`%s @> '"%s"'`, field, value))
+				} else {
+					q = q.Where(fmt.Sprintf(`%s @> '{"%s":"%s"}'`, field, innerField, value))
```

Comment:
Will this work for nested traits?
E.g. checking if `traits.name.first == "bob"`?

Reply:
Also, is value in this case only allowed to be of type string? As the JSON type is always double quoted, it would always be a string in JSON.

Reply:
1. Should work I forgot to add test with that
2. in this case as value come from the query it always a string. I assume that could be work for int too. I'll add a test for this like phone number

Reply:
> Will this work for nested traits?
> E.g. checking if `traits.name.first == "bob"`?

it's work